lib/image-stream/handlers.js

```diff
@@ -123,6 +124,46 @@ module.exports = {
   },
 
   /**
+   * @summary Handle Apple disk images (.dmg)
+   * @function
+   * @public
+   * @memberof handlers
+   *
+   * @param {String} file - file path
+   * @param {Object} options - options
+   * @param {Number} [options.size] - file size
+   *
+   * @fulfil {Object} - image metadata
+   * @returns {Promise}
+   */
+  'application/x-apple-diskimage': (file, options) => {
+    return new Bluebird((resolve, reject) => {
+      const dmg = new UDIF.Image(file);
```

Comment:
BTW, `info.size.final.value` always gets replaced, and `info` is only used at the end, so I guess this could be:

```js
const dmg = Bluebird.promisifyAll(new UDIF.Image(file));

return dmg.openAsync().then(() => {
  return dmg.closeAsync().return({
    stream: UDIF.createReadStream(file),
    size: {
      original: options.size,
      final: {
        estimation: false,
        value: dmg.getUncompressedSize()
      }
    },
    transform: new PassThroughStream()
  });
});
```